08-rails/forms-strong-params.md

```diff
@@ -0,0 +1,81 @@
+# Rails Forms: Strong Params
+
+## Learning Goals
+- Identify what strong parameters are and why Rails requires them
+- Extend Rails forms to use strong parameters
+
+### Getting Started
+
+Let's consider where we ended up most recently with our controller method that handled adding a new book, `create`. Here is the code we ended up with:
+```ruby
+# in app/controllers/books_controller.rb
+def create
+  @book = Book.new(author: params[:book][:author], title: params[:book][:title]) #instantiate a new book
+  if @book.save # save returns true if the database insert succeeds
+    redirect_to root_path # go to the index so we can see the book in the list
+  else # save failed :(
+    render :new # show the new book form view again
+  end
+end
+```
+
+This code is written with the assumption that our `params` hash contains data as follows:
+```ruby
+{ book:
+  {
+    author: "some author",
+    title: "some title"
+  }
+}
+```
+
+Notice that the parameters passed into `Book.new` are all listed out individually. What if we wanted to pass all of the data from the `params` directly without listing out each individual field? What happens when we try: `@book = Book.new(params[:book)])`?
+
+What happened? Most likely, you've gotten a **ForbiddenAttributeError**.
+
+In Rails, controllers are **not permitted to pass parameters directly** from the form into our ActiveRecord model instances. This is a security feature to ensure users cannot update sensitive model attributes that they should not be updating.
+
+### Strong Parameters
+Rails has introduced the concept of **strong parameters** to use in our controllers to prevent these errors. Strong params will require us to give explicit access to a set of attributes that come through `params` that we'll then be able to pass to our model methods. We set up strong params once for our controller, and then we will no longer need to specify each field individually (in the way we had to above).
+
+### Update the Controller
+Now that we have an idea of _why_ we need strong params, let's see how to implement these in our controller.
+
+1. Create a `private` method section towards the bottom of our controller.
+2. Create a new method that we'll use to return the _permitted_ parameters.
+    ```ruby
+    private
+
+    def book_params
+      # code here
+    end
+    ```
+3. We'll use two special controller methods here which will allow us to define exactly which parameters (coming from the form) we want to allow. Those methods are `require` and `permit`.
+
+  We use `require` to identify the highest-level param which contains all others and that we cannot live without. We then use `permit` to list the params within that overall one which we want to allow.
+    ```ruby
+    private
+
+    def book_params
+      params.require(:book).permit(:author, :title)
+    end
```

Comment:
Could you use an explicit return here? I think that makes it more obvious what this code is doing.